Dockerfile.template

```
@@ -134,8 +134,12 @@ RUN set -ex; \
 # this filename needs to sort higher than all the architecture filenames ("aarch64-...", "armeabi...", etc)
 	{ echo '/usr/local/lib64'; echo '/usr/local/lib'; } > /etc/ld.so.conf.d/000-local-lib.conf; \
 	ldconfig -v; \
-	# smoke test to make sure the libs from /usr/local don't break Debian
-	apt-get --version
+	# the libc created by gcc might be too old for a newer Debian
+	# check that the Debian libstdc++ doesn't have newer requirements than the gcc one
+	bash -c '[ "$(comm -23 \
+		<(strings /usr/lib/x86_64-linux-gnu/libstdc++.so.6 | grep GLIB | sort) \
+		<(strings /usr/local/lib64/libstdc++.so | grep GLIB | sort) \
+		)" = "" ]'
```

A few minor tweaks for readability, multiarch correctness, and to make sure `pipefail` catches failures appropriately :eyes:

```suggestion
	bash -Eeuo pipefail -xc ' \
		deb="$(strings /usr/lib/*/libstdc++.so* | grep "^GLIBC" | sort -u)"; \
		gcc="$(strings /usr/local/lib*/libstdc++.so | grep "^GLIBC" | sort -u)"; \
		diff="$(comm -23 <(cat <<<"$deb") <(cat <<<"$gcc"))"; \
		test -z "$diff"; \
	'
```

(warning: untested :joy:)

---

(note this test is on the existing and definitely known-broken `gcc:9` image that's bookworm-based)

```console
$ docker run -it --rm --pull=always gcc:9
9: Pulling from library/gcc
Digest: sha256:caed9ef87573ba4bab85f119b8c7dbbb0f07ff9ec901553b35e62bc778ff69bd
Status: Image is up to date for gcc:9
root@72b3b931ed98:/# bash -Eeuo pipefail -xc ' \
                deb="$(strings /usr/lib/*/libstdc++.so* | grep "^GLIBC" | sort -u)"; \
                gcc="$(strings /usr/local/lib*/libstdc++.so | grep "^GLIBC" | sort -u)"; \
                diff="$(comm -23 <(cat <<<"$deb") <(cat <<<"$gcc"))"; \
                test -z "$diff"; \
        '
++ grep '^GLIBC'
++ sort -u
++ strings /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30
+ deb='GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.16
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_3.4.2
GLIBCXX_3.4.20
GLIBCXX_3.4.21
GLIBCXX_3.4.22
GLIBCXX_3.4.23
GLIBCXX_3.4.24
GLIBCXX_3.4.25
GLIBCXX_3.4.26
GLIBCXX_3.4.27
GLIBCXX_3.4.28
GLIBCXX_3.4.29
GLIBCXX_3.4.3
GLIBCXX_3.4.30
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_DEBUG_MESSAGE_LENGTH
GLIBC_2.14
GLIBC_2.16
GLIBC_2.17
GLIBC_2.18
GLIBC_2.2.5
GLIBC_2.25
GLIBC_2.3
GLIBC_2.3.2
GLIBC_2.32
GLIBC_2.33
GLIBC_2.34
GLIBC_2.36
GLIBC_2.4
GLIBC_2.6'
++ grep '^GLIBC'
++ sort -u
++ strings /usr/local/lib64/libstdc++.so
+ gcc='GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.16
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_3.4.2
GLIBCXX_3.4.20
GLIBCXX_3.4.21
GLIBCXX_3.4.22
GLIBCXX_3.4.23
GLIBCXX_3.4.24
GLIBCXX_3.4.25
GLIBCXX_3.4.26
GLIBCXX_3.4.27
GLIBCXX_3.4.28
GLIBCXX_3.4.3
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_DEBUG_MESSAGE_LENGTH
GLIBC_2.14
GLIBC_2.16
GLIBC_2.17
GLIBC_2.18
GLIBC_2.2.5
GLIBC_2.3
GLIBC_2.3.2
GLIBC_2.33
GLIBC_2.34
GLIBC_2.4
GLIBC_2.6'
+++ cat
++ comm -23 /dev/fd/63 /dev/fd/62
+++ cat
+ diff='GLIBCXX_3.4.29
GLIBCXX_3.4.30
GLIBC_2.25
GLIBC_2.32
GLIBC_2.36'
+ test -z 'GLIBCXX_3.4.29
GLIBCXX_3.4.30
GLIBC_2.25
GLIBC_2.32
GLIBC_2.36'
root@72b3b931ed98:/# echo $?
1
```